machine/v1beta1/types_gcpprovider.go

```diff
@@ -65,6 +65,15 @@ const (
 	ConfidentialComputePolicyDisabled ConfidentialComputePolicy = "Disabled"
 )
 
+type ConfidentialVMTechnology string
```

Comment:
ConfidentialComputePolicy was added as an enum for the case that GCP will add another potential value, see upstream discussion:
https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/809#discussion_r1087644437
Consider adding the new options ("sev", "sev-snp") to the  ConfidentialComputePolicy, in case the user set it to "Enabled" use the default.

Reply:
Thanks for pointing this out @eranco74, I think it's worth considering. I've just brought this discussion topic into the upstream patch as well (see https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1410#issuecomment-2647141893).